### PR TITLE
Change 'alias:' to 'redirect_from:' so aliases work again

### PR DESCRIPTION
**Pull request by mstanleyjones**
@londoncalling @johndmulhausen PTAL, I think this is why our aliases don't work. :)

### CODE REVIEW

**Comment by joaofnfernandes** [comment]
Just for reference, from https://github.com/jekyll/jekyll-redirect-from, it seems that the redirect paths should end with a trailing slash.
I noticed that `compose/link-env-deprecated.md` doesn't have one.

**Comment by mstanleyjones** [comment]
@johndmulhausen let me know that he thought we should still be using the alias functionality rather than redirect_from, and that there might be a bug that is causing the redirect problem that Vicky reported elsewhere. I'd like him to look at it when he comes back from vacation before we decide whether we want to make this change or fix that bug. Maybe he can explain to us why we shouldn't use redirect_from. I think it actually creates the old HTML file with a 302 redirect to the new one, and maybe he thought it would make Jenkins take longer. But let's let him give more details.

File: compose/compose-file.md [hunk 1]
```diff
@@ -581,10 +581,10 @@ In the example below, three services are provided (`web`, `worker`, and `db`), a
         image: mysql
         networks:
           new:
-            aliases:
+            redirect_from:
               - database
           legacy:
-            aliases:
+            redirect_from:
```
Comment (johndmulhausen) [inline compose/compose-file.md]: Erroneous replacement

File: docker-for-windows/menu.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-aliases: []
+redirect_from: []
```
Comment (johndmulhausen) [inline docker-for-windows/menu.md]: Remove

File: engine/getstarted/menu.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-aliases: []
+redirect_from: []
```
Comment (johndmulhausen) [inline engine/getstarted/menu.md]: Remove

**Comment by johndmulhausen** [comment]
Added some comments; will be looking at build times. This might be preferred after all so we stop throwing 404s

**Comment by mstanleyjones** [comment]
@johndmulhausen PTAL, I think this is ready. FYI, here is the magic command I used:

```
find . -type f -name "*.md"|while read i; do sed -i '' -e '1,10 s/aliases:/redirect_from:/' "$i"; done
```

This only replaces `alias:` with `redirect_from:` if it occurs in the first 10 lines of the file.

**Comment by johndmulhausen** [comment]
Made a few nips/tucks for consistency, but this looks great, thank you!

**Pull request merged**